From: Jon Ringle <jon@ringle.org>
Date: Wed, 19 Sep 2012 09:35:40 -0500
Subject: [PATCH] define "__socklen_t_defined"

Port from 1.5.15
---
 usr/include/sys/socket.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr/include/sys/socket.h b/usr/include/sys/socket.h
index fbc7711..56d2d29 100644
--- a/usr/include/sys/socket.h
+++ b/usr/include/sys/socket.h
@@ -152,6 +152,7 @@ struct sockaddr {
 	char            sa_data[14];    /* 14 bytes of protocol address */
 };
 typedef int socklen_t;
+#define __socklen_t_defined 1
 struct msghdr {
 	void *msg_name;
 	int msg_namelen;
