From cc514f4c321a028061df517cbf5950ded8df7042 Mon Sep 17 00:00:00 2001
From: unknown author <unknown.author@example.com>
Date: Tue, 15 Mar 2011 16:38:43 -0500
Subject: [PATCH 2/4] closetime.diff

---
 libntp/clocktime.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/libntp/clocktime.c b/libntp/clocktime.c
index 371859c..3d2bb79 100644
--- a/libntp/clocktime.c
+++ b/libntp/clocktime.c
@@ -24,7 +24,8 @@
  * wide enough to collect most data, while close enough to keep things
  * from getting confused.
  */
-#define	CLOSETIME	(4*60*60)
+//#define	CLOSETIME	(4*60*60)
+#define CLOSETIME	(12*365*24*60*60)
 
 
 int
-- 
1.7.1

